src/proposal.mligo

```diff
@@ -29,6 +29,13 @@ let check_if_proposal_exist (proposal_key, store : proposal_key * storage): prop
     then p
     else (failwith proposal_not_exist : proposal)
 
+[@inline]
+let check_if_proposal_flushed_or_dropped (proposal_key, store : proposal_key * storage): proposal =
+  let p = fetch_proposal (proposal_key, store) in
+  if Set.mem (p.start_level, proposal_key) store.proposal_key_list_sort_by_level
+    then (failwith unstake_invalid_proposal : proposal)
+    else p
```

Comment:
I would suggest to either:
- rename this function to `proposal_is_unstakeable`
- move its definition inside `unstake_vote_one` itself

for the reasons that this is only used by that function and that it throws `unstake_invalid_proposal` which is specific to this use.

Reply:
I've moved this to `unstake_vote_one` itself now.